3party/opening_hours/opening_hours.hpp

```diff
@@ -0,0 +1,566 @@
+/*
+  The MIT License (MIT)
+
+  Copyright (c) 2015 Mail.Ru Group
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <iostream>
+#include <chrono>
+#include <type_traits>
+
+namespace osmoh
+{
+
+class Time
+{
+  enum State
```

Comment:
In this case please, rename these flags to STATE_IS_NOT_TIME, STATE_HAVE_HOURS, STATE_HAVE_MINUTES.

The reason is that c-style enums runs out to parent namespace, so that naming is a try to group them back.